src/lib/replication/repl_dev/solo_repl_dev.cpp

```diff
@@ -0,0 +1,136 @@
+#include <homestore/blkdata_service.hpp>
+#include <homestore/logstore_service.hpp>
+#include <homestore/superblk_handler.hpp>
+#include "common/homestore_assert.hpp"
+#include "replication/repl_dev/solo_repl_dev.h"
+
+namespace homestore {
+SoloReplDev::SoloReplDev(superblk< repl_dev_superblk > const& rd_sb, bool load_existing) : m_rd_sb{rd_sb} {
+    if (load_existing) {
+        logstore_service().open_log_store(LogStoreService::DATA_LOG_FAMILY_IDX, m_rd_sb->data_journal_id, true,
+                                          bind_this(SoloReplDev::on_data_journal_created, 1));
+    } else {
+        m_data_journal =
+            logstore_service().create_new_log_store(LogStoreService::DATA_LOG_FAMILY_IDX, true /* append_mode */);
+        m_rd_sb->data_journal_id = m_data_journal->get_store_id();
+    }
+}
+
+void SoloReplDev::on_data_journal_created(shared< HomeLogStore > log_store) {
+    m_data_journal = std::move(log_store);
+    m_rd_sb->data_journal_id = m_data_journal->get_store_id();
+    m_data_journal->register_log_found_cb(bind_this(SoloReplDev::on_log_found, 3));
```

Comment:
3?

Reply:
this is 3 parameter bind. `bind_this` is a shortcut to 
```bind(<blah, blah >, this, std::placeholders::_1, std::placeholders_2, std::placeholders::_3)```

Reply:
i see.